google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -186,10 +189,21 @@ public Timestamp getReadTimestamp() {
     }
   }
 
+  // Exception class used just to track the stack trace at the point when a session was handed out
+  // from the pool.
+  private final class LeakedSessionException extends Exception {
+    private static final long serialVersionUID = 1451131180314064914L;
+
+    private LeakedSessionException() {
+      super("Session was checked out from the pool at:");
```

Comment:
Oh, that at was intended for the stack trace that will be printed below it (and not for time). But I think adding time is a good idea.